config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/environment/EnvironmentBundleCreator.java

```diff
@@ -52,22 +53,22 @@ public Bundle createEnvironmentBundle(Map<String, String> environmentProperties,
                                           String templatizedBundleFolderPath,
                                           String environmentConfigurationFolderPath,
                                           EnvironmentBundleCreationMode mode,
-                                          String bundleFileName) {
+                                          String bundleFileName,
+                                          String policyBundleName) {
         Bundle environmentBundle = new Bundle();
         environmentBundleBuilder.build(environmentBundle, environmentProperties, environmentConfigurationFolderPath, mode);
 
         setTemplatizedBundlesFolderPath(templatizedBundleFolderPath);
         processDeploymentBundles(
                 environmentBundle,
-                collectFiles(templatizedBundleFolderPath, BUNDLE_EXTENSION).stream().map(f -> new FileTemplatizedBundle(f, new File(bundleFolderPath, f.getName()))).collect(toList()),
+                collectFiles(templatizedBundleFolderPath, mode != PLUGIN ? BUNDLE_EXTENSION : policyBundleName + BUNDLE_EXTENSION).stream().map(f -> new FileTemplatizedBundle(f, new File(bundleFolderPath, f.getName()))).collect(toList()),
                 mode,
                 true);
 
         // write the Environment bundle
         final DocumentBuilder documentBuilder = documentTools.getDocumentBuilder();
         final Document document = documentBuilder.newDocument();
 
-        //ToDo : Need to handle bundle name, Project GroupName and version properly
         Map<String, Pair<Element, BundleMetadata>> bundleElements = bundleEntityBuilder.build(environmentBundle,
                 EntityBuilder.BundleType.ENVIRONMENT, document, bundleFileName, "", null);
```

Comment:
is version not required for environment bundle ?

Reply:
it is required and computed from metadata file i.e bundleFileName contains version info as well